gitlab_test.go

```diff
@@ -0,0 +1,115 @@
+package gitlab
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+var (
+// mux is the HTTP request multiplexer used with the test server.
+	mux *http.ServeMux
+
+// client is the Gitlab client being tested.
+	client *Client
+
+// server is a test HTTP server used to provide mock API responses.
+	server *httptest.Server
+)
+
+// setup sets up a test HTTP server along with a gitlab.Client that is
+// configured to talk to that test server.  Tests should register handlers on
+// mux which provide mock responses for the API method being tested.
+func setup() {
```

Comment:
Instead of using a global `client` and `server` var, I would let this func return those values (as I believe that is much more idiomatic). So it would be something like:

```go
func setup() (*Client, *httptest.Server) {
  // test server
  server := httptest.NewServer(http.NewServeMux())

  // gitlab client configured to use test server
  client := NewClient(nil, "")
  client.SetBaseURL(server.URL)

  return client, server
}
```